runtime/lua/vim/fs.lua

```diff
@@ -334,16 +334,123 @@ function M.find(names, opts)
   return matches
 end
 
+--- Split a Windows path into a prefix and a body, such that the body can be processed like a POSIX
+--- path. Path must use forward slashes (/) as the path separator and must not have any extraneous
+--- slashes.
+---
+--- Does not check if the path is a valid Windows path. Invalid paths will give invalid results.
+---
+--- Examples:
+--- - `//./C:/foo/bar` -> `//./C:`, `/foo/bar`
+--- - `//?/UNC/server/share/foo/bar` -> `//?/UNC/server/share`, `/foo/bar`
+--- - `C:/foo/bar` -> `C:`, `/foo/bar`
+--- - `C:foo/bar` -> `C:`, `foo/bar`
+---
+--- @param path string Path to split.
+--- @return string, string : prefix, body
+local function split_windows_path(path)
+  local prefix = ''
+
+  local function process_unc_path()
+    local server_share = path:match('[^/]+/[^/]+/')
+
+    if server_share == nil then
+      error('Invalid Windows UNC path')
+    end
+
+    prefix = prefix .. server_share --[[ @as string ]]
+    path = path:sub(#server_share + 1)
+  end
+
+  -- Device paths
+  if path:match('^//[?.]/') then
+    prefix = path:sub(1, 4)
+    path = path:sub(5)
```

Comment:
```suggestion
    prefix, path = path:sub(1, 4), path:sub(5)
```

This might be nit picking and against the overall code base, but I usually find it useful having this type of string split as a tuple assignment.